packages/langium/src/workspace/workspace-manager.ts

```diff
@@ -46,22 +50,22 @@ export class DefaultWorkspaceManager implements WorkspaceManager {
     protected readonly mutex: WorkspaceLock;
     protected folders?: WorkspaceFolder[];
 
-    constructor(services: LangiumSharedServices) {
+    constructor(services: LangiumSharedCoreServices) {
         this.serviceRegistry = services.ServiceRegistry;
         this.langiumDocuments = services.workspace.LangiumDocuments;
         this.documentBuilder = services.workspace.DocumentBuilder;
         this.fileSystemProvider = services.workspace.FileSystemProvider;
         this.mutex = services.workspace.WorkspaceLock;
+    }
 
-        services.lsp.LanguageServer.onInitialize(params => {
-            this.folders = params.workspaceFolders ?? undefined;
-        });
+    initialize(params: InitializeParams): void {
+        this.folders = params.workspaceFolders ?? undefined;
+    }
```

Comment:
We explicitly went away from this style of initialization, see https://github.com/eclipse-langium/langium/pull/593. I don't think we should reintroduce it.

Reply:
@montymxb Why is this resolved?

Reply:
We had an internal discussion on that.

Reply:
See https://github.com/eclipse-langium/langium/pull/1258#discussion_r1461918831